File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
       vector<int>result;
        int leftp = 1;
        int rightp = 1;
        for(int i=0;i<nums.size();++i){
            result.push_back(leftp);
            leftp*=nums[i];
        }
        for(int i=nums.size()-1;i>=0;--i){
            int r = rightp * result[i];
            result[i] = r;
            rightp*=nums[i];
        }
        return result;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = []
        leftP= 1
        for i in range(len(nums)):
            result.append(leftP)
            leftP*=nums[i]
        i=len(nums)-1
        rightP = 1
        while i>=0:
            result[i] = result[i] * rightP
            rightP*=nums[i]
            i-=1
        return result